src/sqlfluff/dialects/dialect_clickhouse.py

```diff
@@ -350,6 +350,32 @@ class DatatypeSegment(BaseSegment):
             StringParser("NULLABLE", CodeSegment, type="data_type_identifier"),
             Bracketed(Ref("DatatypeSegment")),
         ),
+        Sequence(
+            # double args
+            Sequence(
+                OneOf(
+                    StringParser("DECIMAL", CodeSegment, type="data_type_identifier"),
+                    StringParser("NUMERIC", CodeSegment, type="data_type_identifier"),
+                ),
+                Ref("BracketedArguments", optional=True),
+                optional=True,
+            ),
+            # single args
+            Sequence(
+                OneOf(
+                    StringParser("DECIMAL32", CodeSegment, type="data_type_identifier"),
+                    StringParser("DECIMAL64", CodeSegment, type="data_type_identifier"),
+                    StringParser(
+                        "DECIMAL128", CodeSegment, type="data_type_identifier"
+                    ),
+                    StringParser(
+                        "DECIMAL256", CodeSegment, type="data_type_identifier"
+                    ),
+                ),
+                OptionallyBracketed(Ref("NumericLiteralSegment")),  # precision
+                optional=True,
+            ),
+        ),
```

Comment:
```suggestion
        # double args
        Sequence(
            OneOf(
                StringParser("DECIMAL", CodeSegment, type="data_type_identifier"),
                StringParser("NUMERIC", CodeSegment, type="data_type_identifier"),
            ),
            Ref("BracketedArguments", optional=True),
        ),
        # single args
        Sequence(
            OneOf(
                StringParser("DECIMAL32", CodeSegment, type="data_type_identifier"),
                StringParser("DECIMAL64", CodeSegment, type="data_type_identifier"),
                StringParser(
                    "DECIMAL128", CodeSegment, type="data_type_identifier"
                ),
                StringParser(
                    "DECIMAL256", CodeSegment, type="data_type_identifier"
                ),
            ),
            Bracketed(Ref("NumericLiteralSegment"), optional=True),  # precision
        ),
```

Reply:
Thanks, I'll restart from this because taking a second look Decimal32 is not even allowed